File: SteamDroid/src/com/flyingpie/steamdroid/app/ChangeState.java
package com.flyingpie.steamdroid.app;

import com.flyingpie.steamdroid.R;
import com.flyingpie.steamdroid.api.State;
import com.flyingpie.steamdroid.util.SteamAdapters;

import android.app.ListActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ListView;

public class ChangeState extends ListActivity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		
		setListAdapter(SteamAdapters.getStateAdapter());
		getListView().setBackgroundColor(getResources().getColor(R.color.listBackground));
	}

	@Override
	protected void onListItemClick(ListView l, View v, int position, long id) {
		super.onListItemClick(l, v, position, id);
		
		State state = SteamAdapters.getStateAdapter().getItem(position);
		state.setActive();
	}
}

File: SteamDroid2/SteamDroid2/App/ChangeState.cs
using Android.App;
using Android.OS;
using Android.Views;
using Android.Widget;

using SteamKit2;

namespace SteamDroid.App
{
	[Activity (Label = "ChangeState")]			
	public class ChangeState : ListActivity
	{
		private ArrayAdapter<EPersonaState> stateAdapter;
		
		protected override void OnCreate(Bundle bundle)
		{
			base.OnCreate(bundle);
			
			stateAdapter = new ArrayAdapter<EPersonaState>(this, Android.Resource.Layout.SimpleListItem1);
			stateAdapter.Add(EPersonaState.Online);
			stateAdapter.Add(EPersonaState.Away);
			stateAdapter.Add(EPersonaState.Busy);
			stateAdapter.Add(EPersonaState.Snooze);
			stateAdapter.Add(EPersonaState.Offline);
			
			ListAdapter = stateAdapter;
		}
		
		protected override void OnListItemClick(ListView l, View v, int position, long id)
		{
			base.OnListItemClick(l, v, position, id);
			
			EPersonaState state = stateAdapter.GetItem(position);
			
			SteamService.GetClient().Friends.SetPersonaState(state);
		}
	}
}

